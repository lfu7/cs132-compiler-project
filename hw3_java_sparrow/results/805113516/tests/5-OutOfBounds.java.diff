2,11c2,3
< Exception in thread "main" java.lang.RuntimeException: "array index out of bounds"
< 	at sparrow.visitor.Interpreter.visit(Interpreter.java:297)
< 	at sparrow.ErrorMessage.accept(ErrorMessage.java:14)
< 	at sparrow.visitor.Interpreter.stepUntilReturn(Interpreter.java:34)
< 	at sparrow.visitor.Interpreter.visit(Interpreter.java:367)
< 	at sparrow.Call.accept(Call.java:19)
< 	at sparrow.visitor.Interpreter.stepUntilReturn(Interpreter.java:34)
< 	at sparrow.visitor.Interpreter.visit(Interpreter.java:97)
< 	at sparrow.Program.accept(Program.java:15)
< 	at Main.main(Main.java:41)
---
> --- Error (Explicit) ---
> array index out of bounds
