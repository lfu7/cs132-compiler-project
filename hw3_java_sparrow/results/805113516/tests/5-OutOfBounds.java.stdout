func _Main()
v2 = 4
v3 = 4
v1 = alloc(v2)
vmt_address4 = alloc(v3)
[v1 + 0] = vmt_address4
f5 = @A_run
[vmt_address4 + 0] = f5
vmt_address6 = [v1 + 0]
f7 = [vmt_address6 + 0]
v0 = call f7(v1)
print(v0)
      return v0

func A_run(this)
v1 = 20
v3 = 4
v2 = v1 * v3
v2 = v2 + v3
v0 = alloc(v2)
[v0 + 0] = v1
a = v0
v5 = a
v6 = 10
v7 = [v5 + 0]
v8 = v6 < v7
if0 v8 goto if_else9
goto if_end10
if_else9:
error("array index out of bounds")
if_end10:
v11 = 4
v12 = v6 * v11
v12 = v12 + v11
v13 = v5 + v12
v4 = [v13 + 0]
print(v4)
v15 = a
v16 = 40
v17 = [v15 + 0]
v18 = v16 < v17
if0 v18 goto if_else19
goto if_end20
if_else19:
error("array index out of bounds")
if_end20:
v21 = 4
v22 = v16 * v21
v22 = v22 + v21
v23 = v15 + v22
v14 = [v23 + 0]
      return v14


